### PR TITLE
#159965269 Implement articles CRUD

### PR DESCRIPTION
**Pull request by micahoriaso**
#### What does this PR do?
Implement articles crud

#### Description of tasks to be completed
- User should be able to create an article
- User should be able to update an article
- User should be able to delete an article
- User should be able to view all articles
- User should be able to view a single article

#### How should this be manually tested?
$ git clone https://github.com/andela/mag-ah-frontend.git
$ git checkout ft-articles-crud-159965269
$ cd mag-ah-frontend
$ npm install
$ npm start

#### What are the relevant pivotal tracker stories?
[#159965269](https://www.pivotaltracker.com/story/show/159965269t)

##### All articles
screen shot 2018-10-26 at 10 05 27

##### Single Article
screen shot 2018-10-26 at 10 08 34

##### New Article
screen shot 2018-10-26 at 10 09 20

##### Update Article
screen shot 2018-10-26 at 10 10 28

##### My Articles also showing edit and delete
screen shot 2018-10-26 at 10 11 32

### CODE REVIEW

**Comment by Bileonaire** [comment]
Looks good to me

File: src/components/articles/MyArticles.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import PropTypes from "prop-types";
+import jwtDecode from "jwt-decode";
+import { fetchArticles } from "../../redux/actions/fetchArticles";
+import { deleteArticle } from "../../redux/actions/deleteArticle";
+import FlatArticle from "../../views/Article/FlatArticle";
+import { clearState } from "../../redux/actions/common";
+
+class Articles extends Component {
+  constructor(props) {
+    super(props);
```
Comment (kimotho-njoki) [inline src/components/articles/MyArticles.jsx]: Is this constructor necessary?

File: src/components/articles/MyArticles.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    articles: []
+  };
+  const { success } = deleteArticleReducer || { success: false };
+  return {
+    articles,
+    success
+  };
+};
+
+export default connect(
```
Comment (kimotho-njoki) [inline src/components/articles/MyArticles.jsx:78]: Refactor to
export default connect(mapStateToProps, {
fetch: blah,
trash: blah,
reset: blah
})(Articles);

File: src/components/articles/UpdateArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                  error={error.description}
+                  required
+                />
+                <TextArea
+                  placeholder="Write your story"
+                  name="body"
+                  value={body}
+                  onChange={this.handleChange}
+                  className={
+                    error.body
```
Comment (kimotho-njoki) [inline src/components/articles/UpdateArticle.jsx:100]: Nice job on the arrangement. Improvement.

File: src/redux/action_types/index.js [hunk 4]
```diff
@@ -17,3 +17,17 @@ export const SOCIAL_LOGIN_START = "SOCIAL_LOGIN_START";
 export const SOCIAL_LOGIN_SUCCESS = "SOCIAL_LOGIN_SUCCESS";
 export const SOCIAL_LOGIN_FAIL = "SOCIAL_LOGIN_FAIL";
 export const RESET_STATE = "RESET_STATE";
+// Article action types
```
Comment (kimotho-njoki) [inline src/redux/action_types/index.js]: Unnecessary comment. Delete it. Remove the rest as well in this file. By looking at the constant I can understand what it is for.

File: src/redux/actions/fetchArticles.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import config from "../../config";
+
+const url = `${config.BASE_URL}/articles/`;
+
+/**
+ * Action for start article fetch
+ *
+ * @return {object} type
+ */
+export const startArticleFetch = () => ({
```
Comment (kimotho-njoki) [inline src/redux/actions/fetchArticles.js]: Why import `startFetch()` in all the previous files but in this instance create an entire action creator?

File: src/components/articles/MyArticles.jsx (hunk 2 above)
Comment (micahoriaso) [reply src/components/articles/MyArticles.jsx:78]: Fixed

File: src/components/articles/UpdateArticle.jsx (hunk 3 above)
Comment (micahoriaso) [reply src/components/articles/UpdateArticle.jsx:100]: Thank you!

File: src/redux/actions/fetchArticles.js (hunk 5 above)
Comment (micahoriaso) [reply src/redux/actions/fetchArticles.js]: Fixed

File: src/redux/action_types/index.js (hunk 4 above)
Comment (micahoriaso) [reply src/redux/action_types/index.js]: Fixed

File: src/components/articles/MyArticles.jsx (hunk 1 above)
Comment (micahoriaso) [reply src/components/articles/MyArticles.jsx]: Fixed

File: src/redux/actions/viewSingleArticle.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import axios from "axios";
+import { VIEW_ARTICLE } from "../action_types";
+import { serverError } from "./common";
+import config from "../../config";
+
+const baseUrl = config.BASE_URL;
+
+const viewSingleArticle = slug => async dispatch => {
+  /**
```
Comment (kimotho-njoki) [inline src/redux/actions/viewSingleArticle.js:9]: There should always be an empty line before a comment.

File: src/redux/actions/viewSingleArticle.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+   * Successfull single article fetch
+   *
+   * @param (object) article
+   * @return (object) type and payload
+   */
+
```
Comment (kimotho-njoki) [inline src/redux/actions/viewSingleArticle.js:15]: This extra line should not be here. No extra line after a comment.

File: src/redux/actions/viewSingleArticle.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import axios from "axios";
+import { VIEW_ARTICLE } from "../action_types";
+import { serverError } from "./common";
+import config from "../../config";
+
+const baseUrl = config.BASE_URL;
+
+const viewSingleArticle = slug => async dispatch => {
+  /**
+   * Successfull single article fetch
```
Comment (kimotho-njoki) [inline src/redux/actions/viewSingleArticle.js:10]: Is this how successful is spelled?

**Pull request merged**